tensorflow_datasets/core/download/kaggle.py

```diff
@@ -49,156 +49,92 @@
-        "kaggle",
-        self._kaggle_type.download_cmd,
-        "files",
-        "-v",
-        self._competition_name,
-    ]
-    output = _run_kaggle_command(command, self._competition_name)
-    return sorted([
-        line.split(",")[0] for line in output.split("\n")[1:] if line
-    ])
-
-  @utils.memoized_property
-  def competition_urls(self):
-    """Returns 'kaggle://' urls."""
-    return [
-        KaggleFile(self._competition_name, fname).to_url()
-        for fname in self.competition_files  # pylint: disable=not-an-iterable
-    ]
-
-  def download_file(self, fname, output_dir):
-    """Downloads competition file to output_dir."""
-    if fname not in self.competition_files:  # pylint: disable=unsupported-membership-test
-      raise ValueError("%s is not one of the competition's "
-                       "files: %s" % (fname, self.competition_files))
-    command = [
-        "kaggle",
-        self._kaggle_type.download_cmd,
-        "download",
-        "--file",
-        fname,
-        "--path",
-        output_dir,
-        self._kaggle_type.dl_flag,
-        self._competition_name
-    ]
-    if self._kaggle_type.extra_flag:
-      command.append(self._kaggle_type.extra_flag)
-    _run_kaggle_command(command, self._competition_name)
-    # kaggle silently compresses some files to '.zip` files.
-    # TODO(tfds): use --unzip once supported by kaggle
-    # (https://github.com/Kaggle/kaggle-api/issues/9)
-    fpath = os.path.join(output_dir, fname + ".zip")
-    if zipfile.is_zipfile(fpath):
-      e = extractor.get_extractor()
-      with e.tqdm():
-        e.extract(fpath, resource.ExtractMethod.ZIP, output_dir).get()
-    return os.path.join(output_dir, fname)
+def is_kaggle_url(url):
+  return url.startswith("kaggle.com")
+
+
+def kaggle_dir_name(competition_or_dataset):
+  """Returns name of dir where the dataset is to be downloaded."""
+  return competition_or_dataset.replace("/", "_")
+
+
+def get_kaggle_url(competition_or_dataset):
+  """Returns 'kaggle.com' urls."""
+  return "%s/%s" % ("kaggle.com", competition_or_dataset)
+
+
+def log_command_output(output, error=False):
+  log = logging.error if error else logging.info
+  log("kaggle command output:\n%s", tf.compat.as_text(output))
 
 
-def _run_kaggle_command(command_args, competition_name):
+def run_kaggle_command(command_args, competition_or_dataset):
   """Run kaggle command with subprocess."""
   try:
     output = sp.check_output(command_args)
     return tf.compat.as_text(output)
   except sp.CalledProcessError as err:
     output = err.output
-    _log_command_output(output, error=True)
+    log_command_output(output, error=True)
     if output.startswith(b"404"):
-      logging.error(_NOT_FOUND_ERR_MSG, competition_name)
+      logging.error(_NOT_FOUND_ERR_MSG, competition_or_dataset)
       raise
-    logging.error(_ERR_MSG, competition_name)
+    logging.error(_ERR_MSG, competition_or_dataset)
     raise
-  except FileNotFoundError as err:
```

Comment:
Why removing this ?

Reply:
It was removed a long time ago and I'm not sure what the reason was. Adding it back.